source/exe/main_common.cc

```diff
@@ -40,11 +40,25 @@ Runtime::LoaderPtr ProdComponentFactory::createRuntime(Server::Instance& server,
   return Server::InstanceUtil::createRuntime(server, config);
 }
 
-MainCommonBase::MainCommonBase(OptionsImpl& options) : options_(options) {
+MainCommonBase::MainCommonBase(OptionsImpl& options)
+    : MainCommonBase(options, nullptr, nullptr, nullptr,
+                     std::unique_ptr<Runtime::RandomGenerator>()) {}
+
+MainCommonBase::MainCommonBase(OptionsImpl& options, Event::TimeSystem* time_system,
```

Comment:
Could we just have a single constructor and make the caller pass the various things by reference? Just curious if we can simplify here vs. having multiple constructors?

Reply:
Sure.  That means specifying defaults in two places (MainCommon and main_common both consume MainCommonBase) but it also means we can completely get rid of the conditional allocation, which on balance seems like a win.  Will do.